src/containers/BooksList.js

```diff
@@ -0,0 +1,42 @@
+import React          from 'react';
+import {connect}      from 'react-redux';
+import Book           from '../components/Book';
+import { removeBook, changeFilter } from '../actions/index';
+import CategoryFilter from "../components/CategoryFilter";
+
+function mapStateToProps(state) {
+  return {books: state.books, filter: state.filter};
+}
+
+const mapDispatchToProps = dispatch => ({
+    remove: (book) => dispatch(removeBook(book)),
+    changeFilter: (filter) => dispatch(changeFilter(filter)),
+});
+
+const BooksList = (props) => {
+  const {books, filter} = props;
+
+  const handleRemoveBook = (book) =>{
+    props.remove(book);
+  };
+
+  const handleFilterChange = (filter) => {
+    props.changeFilter(filter);
+  };
+
+  let showBooks = filter === 'All' ? books : books.filter(book => book.category === filter);
+
+  return (
+    <div className="bookList">
+      <CategoryFilter onFilterChange={(e) => handleFilterChange(e.target.value)} />
+        {showBooks.map((book) => {
+          return <Book key={book.id} book={book} onRemove={()=>handleRemoveBook(book)} />
```

Comment:
No reason to pass both `book.id` and `book`. ID should be an attribute of the book object.

Reply:
React is asking for the distinctive key, that is the only reason we passed book id.

Reply:
Oh sorry, I missed that. All good then :ok: